src/Servers/IIS/IIS/test/Common.FunctionalTests/StartupTests.cs

```diff
@@ -1243,6 +1267,7 @@ private async Task WebConfigOverridesHostingStartup(HostingModel hostingModel)
         public Task WebConfigExpandsVariables_InProcess() => WebConfigExpandsVariables(HostingModel.InProcess);
 
         [ConditionalFact]
+        [RequiresNewShim]
```

Comment:
So if I'm reading this right, this change no longer runs out-of-proc startup and shutdown tests with AspNetCoreModuleV2 2.2.0. Is that right?

Is the idea that in-proc is more likely to regress than out-of-proc? Do we run any out-of-proc tests with AspNetCoreModuleV2 2.2.0 after this change?

Reply:
Good questions.

So yes, this change removes OutOfProcess running with ANCMV2 2.2. This means on every test run, we will not be running an older version of ANCM against a newer version of .NET. There are a few reasons I believe this is fine.

- We are still running all out of proc tests with both a new and old version of .NET, just always on the latest global components. This will still catch regressions that occur in .NET as well as regressions in any new changes to ANCM.
- In-proc is significantly more likely to regress because the interactions between shim and handler happen in native code, rather than with HTTP requests.
- It's hard to imagine a scenario where something would regress out-of-proc in 2.2 versus out-of-proc in master/5.0
- Out-of-proc has been fairly stable over the last few years, not receiving many updates.

For sanity sake, I can keep one ANCMV2 2.2 test with out of proc. Would that make this change more reasonable?

Reply:
I think having one test which covers ANCMV2 2.2 out-of-proc is a good idea. I agree with the premise that regressions are less likely out-of-proc compared to in-proc, so one should be sufficient.

Reply:
Added one back.